src/circadian_model.R

```diff
@@ -1,162 +1,226 @@
-# Here we will have all the functions related to the circadian model
-# We will be implementing Hannay's 2019 single population model plus
-# the sleep drive model, just like you have in your code
 library(deSolve)
-library(tidyverse)
-
-# Model parameters
-Dwake <-555.4
-Dsleep <- 572.7 
-# TODO: add parameters
-alpha0 <- 0.05
-p <- 1.5
-I0 <- 9325.0
-Delta <- 0.0075
-G <- 33.75
-
-Irecep <- 1 / 291.0
-Targc <- 0.9677
-Tau <- 23.84
-Beta1 <- -0.09318
-A1    <- 0.3855
-A2    <- 0.1977
-BetaL1<- -0.0026 
-BetaL2<- -0.957756
-Sigma <- 0.0400692
-K     <- 0.06458
-Gamma <- 0.024
-
-# Define parameters vector
-sleepParameters <- c(Dwake, Dsleep)
-lightParameters <- c(alpha0, p, I0, Delta, G)
-circadianParameters <- c(Irecep, Targc, Tau, Beta1, A1, A2, BetaL1, BetaL2, Sigma, K, Gamma)
-# Model initial conditions
-# TODO: add initial conditions
-# ...
-default_initial_conditions <- c(R=0.8240484019863923, Psi=2.4532409247791804, n=0.78424752754260862, A=767.657, R1tot=584.24)
-
-chiAwake <- 1.0/18.18
-muAwake <- 869.5
-chiAsleep <- 1.0/7.0
-muAsleep <- 596.5
-
-# params <- c(Dwake, Dsleep, alpha0, p, I0, Delta, G, Irecep, Targc, Tau, Beta1, A1, A2, BetaL1, BetaL2, Sigma, K, Gamma, chiAwake, muAwake, chiAsleep, muAsleep)
-
-pow <- function(a, p) {
-  a ** p 
-}
+library(tidyVerse)
+source(file.path(“test/test_model.R”), local = TRUE)$value 
+
+schoolInfo <- c((t %% 24 > params[["schoolStart"]]) && (t %% 24 <= (params[["schoolStart"]] + params[["schoolDuration"]])), ((t %/% 24) %% 7) < 5)
```

Comment:
Your implementation makes sense. However, this line is what we want to avoid. We don't want global variables because we expect `params` to change every time the users changes the input. Therefore, we can't have it defined here. A possible solution is to have `schoolInfo` within typicalLight`